pkg/admission/apibinding/apibinding_admission.go

```diff
@@ -0,0 +1,225 @@
+	}
+	apiBinding, ok := obj.(*apisv1alpha1.APIBinding)
+	if !ok {
+		// nolint: nilerr
+		return nil // only work on unstructured APIBindings
+	}
+
+	// Return early if there's nothing to validate
+	if apiBinding.Spec.Reference.Workspace == nil {
+		return nil
+	}
+
+	// Object validation
+	var errs field.ErrorList
+
+	switch a.GetOperation() {
+	case admission.Create:
+		errs = ValidateAPIBinding(apiBinding)
+	case admission.Update:
+		obj, err = kcpadmissionhelpers.NativeObject(a.GetOldObject())
+		if err != nil {
+			return fmt.Errorf("unexpected unknown old object, got %v, expected APIBinding", a.GetOldObject().GetObjectKind().GroupVersionKind().Kind)
+		}
+		old, ok := obj.(*apisv1alpha1.APIBinding)
+		if !ok {
+			return fmt.Errorf("unexpected unknown old object, got %v, expected APIBinding", obj.GetObjectKind().GroupVersionKind().Kind)
+		}
+		errs = ValidateAPIBindingUpdate(old, apiBinding)
+	}
+	if len(errs) > 0 {
+		return admission.NewForbidden(a, fmt.Errorf("%v", errs))
+	}
+
+	// Determine the cluster name for the referenced export
+	cluster, err := genericapirequest.ValidClusterFrom(ctx)
+	if err != nil {
+		return admission.NewForbidden(a, fmt.Errorf("error determining workspace: %w", err))
+	}
+	org, _, err := helper.ParseLogicalClusterName(cluster.Name)
+	if err != nil {
+		return admission.NewForbidden(a, fmt.Errorf("%q is not a valid workspace name: %w", cluster.Name, err))
+	}
+	apiExportClusterName := helper.EncodeOrganizationAndClusterWorkspace(org, apiBinding.Spec.Reference.Workspace.WorkspaceName)
+
+	// Access check
+	if err := o.checkAPIExportAccess(ctx, a.GetUserInfo(), apiExportClusterName, apiBinding.Spec.Reference.Workspace.ExportName); err != nil {
+		action := "create"
+		if a.GetOperation() == admission.Update {
+			action = "update"
+		}
+		return admission.NewForbidden(a, fmt.Errorf("unable to %s APIImport: %w", action, err))
+	}
+
+	return nil
+}
+
+func (o *apiBindingAdmission) checkAPIExportAccess(ctx context.Context, user user.Info, clusterName, exportName string) error {
+	authz, err := o.createAuthorizer(clusterName, o.kubeClusterClient)
+	if err != nil {
+		// Logging a more specific error for the operator
+		klog.Errorf("error creating authorizer from delegating authorizer config: %v", err)
+		// Returning a less specific error to the end user
+		return errors.New("unable to authorize request")
+	}
+
+	bindAttr := authorizer.AttributesRecord{
+		User:            user,
+		Verb:            "bind",
+		APIGroup:        apisv1alpha1.SchemeGroupVersion.Group,
+		APIVersion:      apisv1alpha1.SchemeGroupVersion.Version,
+		Resource:        "apiexports",
+		Name:            exportName,
+		ResourceRequest: true,
+	}
+
+	if decision, _, err := authz.Authorize(ctx, bindAttr); err != nil {
+		return fmt.Errorf("unable to determine access to apiexports: %w", err)
+	} else if decision != authorizer.DecisionAllow {
+		return errors.New("missing verb='bind' permission on apiexports")
```

Comment:
nit: do we have some uniform error string for these kind verb denies?

Reply:
Not at the moment. Are you thinking we should make a custom error type?